src/app/clusters/general_diagnostics_server/general_diagnostics_server.cpp

```diff
@@ -67,6 +72,32 @@ CHIP_ERROR GeneralDiagosticsAttrAccess::ReadIfSupported(CHIP_ERROR (PlatformMana
     return aEncoder.Encode(data);
 }
 
+template <typename T>
+CHIP_ERROR GeneralDiagosticsAttrAccess::ReadListIfSupported(CHIP_ERROR (PlatformManager::*getter)(T &),
+                                                            AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    T faultList;
+
+    if ((PlatformMgr().*getter)(faultList) == CHIP_NO_ERROR)
+    {
+        err = aEncoder.EncodeList([&faultList](const TagBoundEncoder & encoder) -> CHIP_ERROR {
+            for (auto fault : faultList)
+            {
+                ReturnErrorOnFailure(encoder.Encode(fault));
+            }
+
+            return CHIP_NO_ERROR;
+        });
+    }
+    else
+    {
+        err = aEncoder.Encode(DataModel::List<EndpointId>());
```

Comment:
Why do we encode a list of `EndpointId`s?

Reply:
To pas CI check, all attributes enabled in controller clusters are tested via a read in the darwin tests, It's just a simple success/failure test.

Reply:
good catch, it should be unit8_t. Thanks!